src/Sign.Core/DataFormatSigners/ClickOnceSigner.cs

```diff
@@ -126,8 +127,8 @@ await Parallel.ForEachAsync(files, _parallelOptions, async (file, state) =>
 
                     // Inner files are now signed
                     // now look for the manifest file and sign that if we have one
-
-                    FileInfo? manifestFile = filteredFiles.SingleOrDefault(f => ".manifest".Equals(f.Extension, StringComparison.OrdinalIgnoreCase));
+                    var appManifestFromDeploymentManifest = GetApplicationManifestForDeploymentManifest(file);
+                    FileInfo? manifestFile = filteredFiles.SingleOrDefault(f => f.Name.Equals(appManifestFromDeploymentManifest, StringComparison.OrdinalIgnoreCase));
```

Comment:
This is avoidable if `appManifestFromDeploymentManifest` is null.

Also, can we rename `manifestFile` to `applicationManifestFile` since now we're dealing with potentially many different kinds of manifest files.  We had deployment and application manifests, but now we're dealing with other manifests (like assembly manifests).